[DOCS] Adds 9.4.2 Kibana release notes (#270302)
elastic/kibana: docs/release-notes/index.md

% FEATURES, ENHANCEMENTS, FIXES
% Paste in index.md

## 9.4.2 [kibana-9.4.2-release-notes]

### Features and enhancements [kibana-9.4.2-features-enhancements]

**Search**:
* Improves the **Add inference endpoint** and **Edit inference endpoint** flyouts with clearer titles, a footer **Save** button, task type descriptions in the selector, always-visible endpoint ID and API reference fields, and inline validation for required fields [#262143]({{kib-pull}}262143).

### Fixes [kibana-9.4.2-fixes]

**Alerting and cases**:

> * Fixes a bug that caused rules with large action parameter payloads to fail when saving or executing [#269467]({{kib-pull}}269467).

Reviewer: ```suggestion
* Fixes an issue that caused rules with large action parameter payloads to fail when saving or executing [#269467]({{kib-pull}}269467).
```

% FEATURES, ENHANCEMENTS, FIXES
% Paste in index.md

## 9.4.2 [kibana-9.4.2-release-notes]

### Features and enhancements [kibana-9.4.2-features-enhancements]

**Elastic Observability solution**:
For the Elastic Observability 9.4.2 release information, refer to [Elastic Observability Solution Release Notes](docs-content://release-notes/elastic-observability/index.md).

**Elastic Security solution**:
For the Elastic Security 9.4.2 release information, refer to [Elastic Security Solution Release Notes](docs-content://release-notes/elastic-security/index.md).

> 

Reviewer: ```suggestion
```

I find this redundant/low value when we already have them in fixes
